python/setup.py

```diff
@@ -49,64 +52,74 @@
 cuda_include_dir = os.path.join(CUDA_HOME, "include")
 cuda_lib_dir = os.path.join(CUDA_HOME, "lib64")
 
+##############################################################################
+# - Clean target -------------------------------------------------------------
+
+if "clean" in sys.argv:
+    print("Cleaning all Python and Cython build artifacts...")
+
+    treelite_path = ""
+    libcuml_path = ""
+
+    try:
+        shutil.rmtree('build')
+        shutil.rmtree('.pytest_cache', ignore_errors=True)
+        shutil.rmtree('external_repositories', ignore_errors=True)
+        shutil.rmtree('cuml.egg-info', ignore_errors=True)
+        shutil.rmtree('__pycache__', ignore_errors=True)
+
+        clean_folder('cuml')
+
+    except IOError:
+        pass
+
+    # need to terminate script so cythonizing doesn't get triggered after
+    # cleanup unintendedly
+    sys.argv.remove("clean")
+    sys.argv.remove("--all")
+
+    if len(sys.argv) == 1:
+        sys.exit(0)
 
 ##############################################################################
-# - Subrepo checking and cloning ---------------------------------------------
+# - Cloning dependencies if needed -------------------------------------------
 
 subrepos = [
-    'cub',
-    'cutlass',
-    'faiss',
     'treelite'
 ]
 
 # We check if there is a libcuml++ build folder, by default in cpp/build
 # or in CUML_BUILD_PATH env variable. Otherwise setup.py will clone the
-# dependencies defined in cpp/CMakeListst.txt
-if "clean" not in sys.argv:
-    if os.environ.get('CUML_BUILD_PATH', False):
-        libcuml_path = '../' + os.environ.get('CUML_BUILD_PATH')
-    else:
-        libcuml_path = '../cpp/build/'
-
-    found_cmake_repos = get_submodule_dependencies(subrepos,
-                                                   libcuml_path=libcuml_path)
-
-    if found_cmake_repos:
-        treelite_path = os.path.join(libcuml_path,
-                                     'treelite/src/treelite/include')
-        faiss_path = os.path.join(libcuml_path, 'faiss/src/')
-        cub_path = os.path.join(libcuml_path, 'cub/src/cub')
-        cutlass_path = os.path.join(libcuml_path, 'cutlass/src/cutlass')
-    else:
-        # faiss requires the include to be to the parent of the root of
-        # their repo instead of the full path like the others
-        faiss_path = 'external_repositories/'
-        treelite_path = 'external_repositories/treelite/include'
-        cub_path = 'external_repositories/cub'
-        cutlass_path = 'external_repositories/cutlass'
+# dependencies defined in cpp/cmake/Dependencies.cmake
+if os.environ.get('CUML_BUILD_PATH', False):
+    libcuml_path = '../' + os.environ.get('CUML_BUILD_PATH')
+else:
+    libcuml_path = '../cpp/build/'
 
+found_cmake_repos = get_submodule_dependencies(subrepos,
+                                               libcuml_path=libcuml_path)
+
+if found_cmake_repos:
+    treelite_path = os.path.join(libcuml_path,
```

Comment:
this seems very treelite-specific though the variable names are generic (found_cmake_repos)... should it be "found_treelite_repos"?

Reply:
before this PR actually we were optionally cloning a bunch of repos (like faiss). Once there is a treelite conda package we can avoid this actually